great_expectations/expectations/core/unexpected_rows_expectation.py

```diff
@@ -138,7 +141,7 @@ def _validate(
         execution_engine: ExecutionEngine | None = None,
     ) -> Union[ExpectationValidationResult, dict]:
         metric_value = metrics["unexpected_rows_query.table"]
-        unexpected_row_count = len(metric_value)
+        unexpected_row_count = metrics["unexpected_rows_query.row_count"]
```

Comment:
I could refer to `UnexpectedRowsQueryRowCount.metric_name`. It would be a new pattern to use everywhere. We have plans to re-write the metrics engine entirely, so I'm not sure it's worth the effort right now.